### PR TITLE
feat(odm): add DatastoreMixin for enhanced model datastore interaction

### PR DESCRIPTION
**Pull request by timthedevguy-truesec**
### Summary

Introduces a new `DatastoreMixin` class in `api/howler/odm/mixins.py` that gives ODM `Model` subclasses first-class, typed access to their Elasticsearch collection and a `save()` convenience method — eliminating repetitive boilerplate across the codebase. Three core models (`Case`, `Hit`, `Observable`) are updated to use the mixin immediately.

---

#### `api/howler/odm/mixins.py` *(new file)*
- Adds `DatastoreMixin[ModelType]`, a generic mixin providing:
  - **`store`** — a class-level-only descriptor (`_ObjectsDescriptor`) that returns the correctly-typed `ESCollection[ModelType]` for the model. Raises `AttributeError` if accessed from an instance.
  - **`ds`** — an instance property returning the shared datastore connection.
  - **`save()`** — persists the current model instance using its lowercase class name as the index and its configured ID field as the document key.

#### `api/howler/odm/base.py`
- Extends the `@model` decorator with a new `id_field` parameter.
- The decorator now stores `id_field` as `cls._Model__id_field` on the decorated class, defaulting to `<classname_lower>_id` if not specified. This is what `DatastoreMixin.save()` reads to determine the document key.

#### `api/howler/odm/models/case.py`, `hit.py`, `observable.py`
- Each model class now inherits from `DatastoreMixin["<ModelName>"]` in addition to `odm.Model`/`Record`.
- `Case` and `Hit` pass `id_field="howler.id"` to the `@model` decorator to correctly map their primary key (a dotted path).

#### `api/test/unit/odm/test_mixins.py` *(new file, 255 lines)*
- Full unit test coverage for `DatastoreMixin` and `_ObjectsDescriptor`:
  - `TestStoreDescriptor` — class-level access, index name derivation, multi-class isolation, instance-access guard, call-through behaviour, and error message content.
  - `TestDsProperty` — returns datastore, call-through, class-access guard.
  - `TestSaveMethod` — correct index/ID/instance args, return value (`True`/`False`), dotted ID field resolution, and call-count assertions.

---

### Why

Previously, any service or handler that needed to persist a model had to manually resolve the datastore, look up the correct collection, and supply the ID — the same pattern repeated across many call sites. `DatastoreMixin` centralises that logic once, keeps it typed, and makes the intent at each call site obvious (`instance.save()`).

The `id_field` addition to `@model` is a prerequisite: the mixin needs a reliable, declarative way to know which field holds the document's primary key, including dotted-path fields like `howler.id`.

---

### Validation

- 255-line unit test suite covering all three public APIs of the mixin (`store`, `ds`, `save`) across normal, error, and edge-case paths.
- Existing model tests continue to pass (no breaking changes to `@model` decorator — `id_field` defaults gracefully).

**Recommended CI checks:**
```bash
cd api
poetry run test test/unit/odm/test_mixins.py
poetry run ruff check howler --output-format=github
poetry run type_check
```

---

### Risk / Impact

- **Low risk.** The `@model` decorator change is fully backwards-compatible (`id_field` defaults to `<classname_lower>_id`).
- Mixin inheritance is additive; no existing public APIs are removed or changed.
- `_ObjectsDescriptor` actively prevents accidental instance-level access, reducing misuse risk.

---

### Notes

- `DatastoreMixin` is intentionally generic (`Generic[ModelType]`) so type-checkers can infer `Case.store` as `ESCollection[Case]` etc.
- `save()` uses `operator.attrgetter` to support dotted ID paths (e.g. `howler.id`).
- No config, migration, or infrastructure changes required.

### CODE REVIEW

File: api/howler/odm/mixins.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+ModelType = TypeVar("ModelType", bound=Model)
+
+
+class _ObjectsDescriptor(Generic[ModelType]):
+    """Descriptor that provides class-level-only access to the model's ESCollection.
+
+    Intended to be accessed exclusively via the class (e.g. ``Case.store``).
+    Raises ``AttributeError`` if accessed from an instance to enforce
+    cla  # noqa: D205
```
Comment (cccs-mdr) [inline api/howler/odm/mixins.py:27]: Not sure what;s up with this line? A cut-off class description maybe?

**Review by cccs-mdr — APPROVED** [review]

File: api/howler/odm/mixins.py (hunk 1 above)
Comment (timthedevguy-truesec) [reply api/howler/odm/mixins.py:27]: oh yep, I'll fix in a new PR
